ecosystem/typescript/sdk/scripts/publish_ans_contract.ts

```diff
@@ -0,0 +1,62 @@
+const { execSync } = require("child_process");
+require("dotenv").config();
+
+/**
+ * TS SDK supports ANS. Since ANS contract is not part of aptos-framework
+ * we need to get the ANS contract, publish it to local testnet and test against it.
+ * This script clones the aptos-names-contracts repo {@link https://github.com/aptos-labs/aptos-names-contracts},
+ * uses a pre created account address and private key to fund that account and
+ * then publish the contract under that account.
+ * After the contract is published, we delete the cloned repo folder.
+ *
+ * This script runs when testing locally and on CI (as part of sdk-release.yaml) using `pnpm test`.
+ */
+
+// on local publishing we want to use `aptos` commnads and on CI we want to use `docker`
+const APTOS_INVOCATION = process.env.APTOS_INVOCATION || "aptos";
+// environment we use when testing
+const APTOS_NODE_URL = process.env.APTOS_NODE_URL;
+const APTOS_FAUCET_URL = process.env.APTOS_FAUCET_URL;
+// ans account we use to publish the contract
+const ANS_TEST_ACCOUNT_PRIVATE_KEY = process.env.ANS_TEST_ACCOUNT_PRIVATE_KEY;
+const ANS_TEST_ACCOUNT_ADDRESS = process.env.ANS_TEST_ACCOUNT_ADDRESS;
```

Comment:
Now that `pnpm test` requires these env vars to be set, could we either add defaults here or add instructions to the README saying that you must set these? I think the defaults option could be simplest.